zkevm-circuits/src/witness/tx.rs

```diff
@@ -1286,13 +1286,18 @@ pub(super) fn tx_convert(
             .iter()
             .map(|step| step_convert(step, tx.block_num))
             .chain({
-                let rw_counter = tx.steps().last().unwrap().rwc.0 + 9 - (id == 1) as usize;
+                // TODO: it is a bit counter-intuitive to treat EndInnerBlock step, even multiple
+                // EndInnerBlock steps to belong to the last prev tx.
+                // We can change design later to make it easier to understand.
+                let last_step = tx.steps().last().unwrap();
+                let rw_counter = last_step.rwc.0 + last_step.bus_mapping_instance.len();
                 debug_assert!(next_block_num >= tx.block_num);
                 (tx.block_num..next_block_num)
                     .map(|block_num| ExecStep {
                         rw_counter,
                         execution_state: ExecutionState::EndInnerBlock,
                         block_num,
+                        call_index: last_step.call_index,
```

Comment:
why set this to last_step's call_index?

Reply:
I require call_id to be same between EndTx <=> EndInnerBlock (call_id is used to fetch tx_id from call_context).

And to assign call_id correctly for ExecState, a correct call_index is needed.

<img width="859" alt="image" src="https://github.com/scroll-tech/zkevm-circuits/assets/2833376/efafcf42-836a-4808-b8ad-6007b9d01da2">